streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1293,6 +1306,49 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
       ZIO.suspendSucceed(interpret(exec.run().asInstanceOf[ChannelState[Env, OutErr]]))
     }
 
+  final def toPullInAlt(
+    scope: => Scope
+  )(implicit trace: Trace): ZIO[Env, Nothing, ZIO[Env, Either[OutErr, OutDone], OutElem]] =
+    ZIO.uninterruptible {
+      val exec = new ChannelExecutor[Env, InErr, InElem, InDone, OutErr, OutElem, OutDone](
+        () => self,
+        null,
+        identity[URIO[Env, Any]]
+      )
+      for {
+        environment <- ZIO.environment[Env]
+        scope       <- ZIO.succeed(scope)
```

Comment:
Is the suspension necessary?

Reply:
It is a convention, every time the parameter is passed by value, though in this instance I wonder why that decision was made 🤔

Reply:
I don't really understand how that convention came to be, but I don't like it 😅 I think we can remove it (since it's only used once in the effect) and optimize it a bit more as:

```scala
      ZIO.environmentWithZIO[Env] { environment =>
        scope.addFinalizerExit { exit =>
          val finalizer = exec.close(exit)
          if (finalizer ne null) finalizer.provideEnvironment(environment)
          else ZIO.unit
        }
      } *> Exit.succeed(exec)
```

Reply:
I misunderstood, yeah I'll do it